titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/job/JobFederationTest.java

```diff
@@ -106,4 +121,51 @@ public void testJobCreateRouting() {
         assertThat(cell2TaskContext).containsEntry("titus.stack", federatedStackName);
         assertThat(cell2TaskContext).containsEntry("titus.cell", "v3OnlyCell");
     }
+
+    @Test(timeout = LONG_TEST_TIMEOUT_MS)
+    public void testLastSchedulingResult() throws InterruptedException {
+        // Ask for a lot of
+        JobDescriptor<BatchJobExt> tooBigJob = oneTaskBatchJobDescriptor().but(jd -> jd.getContainer().toBuilder()
+                .withContainerResources(ContainerResources.newBuilder()
+                        .withCpu(64)
+                        .withGpu(16)
+                        .withMemoryMB(472_000)
+                        .withDiskMB(999_000)
+                        .withNetworkMbps(40_000)
+                        .build()
+                )
+                .build()
+        );
+        blockingJobClient.createJob(toGrpcJobDescriptor(tooBigJob.toBuilder().withCapacityGroup("a123").build())).getId();
+        blockingJobClient.createJob(toGrpcJobDescriptor(tooBigJob.toBuilder().withCapacityGroup("b123").build())).getId();
+
+        List<Task> tasks = eventStreamObserver.toObservable()
+                .filter(event -> event.getNotificationCase() == JobChangeNotification.NotificationCase.TASKUPDATE)
+                .map(e -> e.getTaskUpdate().getTask())
+                .take(2)
+                .toList()
+                .toBlocking()
+                .first();
+
+        checkSchedulingResult(tasks.get(0));
+        checkSchedulingResult(tasks.get(1));
+    }
+
+    private void checkSchedulingResult(Task task) {
+        while (true) {
```

Comment:
add timeout? Might be easier to use `Awaitability` like other tests do

Reply:
I used `Awaitability` first, but then switched to an explicit loop. I purposefully did not set any timeouts, as it is difficult to estimate the time budget for all the blocking calls we have here. Instead we depend in this test on one global (and quite long) timeout.

Reply:
On a second thought, having fine grain timeouts is better. Adding them...